### PR TITLE
Migrate to null safety and Flame v1rc8

### PR DESCRIPTION
**Pull request by Akida31**
Flame_tiled can now be used from Flame v1 release candidates!

# TODO
update version of `tiled.dart` after it's null safe (for example through [this PR](https://github.com/flame-engine/tiled.dart/pull/19))

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -23,22 +28,22 @@ class TiledGame extends BaseGame {
...
+          SpriteAnimationData.sequenced(
+            amount: 8,
+            textureSize: Vector2.all(20),
+            stepTime: 0.15,
+          ),
         ),
       );
-      comp.x = obj.x.toDouble();
-      comp.y = obj.y.toDouble();
+      comp.x = obj.x!.toDouble();
```
Comment (Schnurber) [inline example/lib/main.dart]: With my new version for tiled, the `!` is not needed.

File: example/lib/main.dart [hunk 2]
```diff
@@ -23,22 +28,22 @@ class TiledGame extends BaseGame {
...
+      comp.y = obj.y!.toDouble();
```
Comment (Schnurber) [inline example/lib/main.dart]: `!` not necessary

File: lib/tiled.dart [hunk 3]
```diff
@@ -88,7 +89,7 @@ class Tiled {
 
   Future _load() async {
     map = await _loadMap();
-    image = await Flame.images.load(map.tilesets[0].image.source);
+    image = await Flame.images.load(map.tilesets[0].image!.source!);
```
Comment (Schnurber) [inline lib/tiled.dart]: for new tiled: `image = await Flame.images.load(map.tilesets[0].image!.source);`

File: lib/tiled.dart [hunk 4]
```diff
@@ -114,20 +115,20 @@ class Tiled {
...
-      batches[batch].clear();
+      batches[batch]!.clear();
     }
     _drawTiles(map);
   }
 
   void _drawTiles(TileMap map) {
-    map.layers.where((layer) => layer.visible).forEach((layer) {
-      layer.tiles.forEach((tileRow) {
+    map.layers.where((layer) => layer.visible!).forEach((layer) {
```
Comment (Schnurber) [inline lib/tiled.dart]: `layer.visible` without `!`

File: lib/tiled.dart [hunk 5]
```diff
@@ -114,20 +115,20 @@ class Tiled {
...
+      layer.tiles!.forEach((tileRow) {
```
Comment (Schnurber) [inline lib/tiled.dart]: remove `!`

File: lib/tiled.dart [hunk 6]
```diff
@@ -114,20 +115,20 @@ class Tiled {
...
         tileRow.forEach((tile) {
-          if (tile.gid == 0) {
+          if (tile!.gid == 0) {
```
Comment (Schnurber) [inline lib/tiled.dart]: remove `!`

File: lib/tiled.dart [hunk 7]
```diff
@@ -138,20 +139,19 @@ class Tiled {
...
           batch.add(
-            rect: src,
-            offset: Offset(
-              tile.x.toDouble() * tileSize.width +
-                  (tile.flips.horizontally ? tileSize.width : 0),
-              tile.y.toDouble() * tileSize.height +
-                  (tile.flips.vertically ? tileSize.height : 0),
+            source: src,
+            offset: Vector2(
+              tile.x!.toDouble() * tileSize.width +
```
Comment (Schnurber) [inline lib/tiled.dart]: remove `!`

File: lib/tiled.dart [hunk 8]
```diff
@@ -138,20 +139,19 @@ class Tiled {
...
+                  (tile.flips!.horizontally ? tileSize.width : 0),
+              tile.y!.toDouble() * tileSize.height +
```
Comment (Schnurber) [inline lib/tiled.dart]: remove `!`

File: lib/tiled.dart [hunk 9]
```diff
@@ -138,20 +139,19 @@ class Tiled {
...
+                  (tile.flips!.vertically ? tileSize.height : 0),
             ),
             rotation: flips.angle * math.pi / 2,
-            scale: tileSize.width / tile.width,
+            scale: tileSize.width / tile.width!,
```
Comment (Schnurber) [inline lib/tiled.dart]: remove `!`

**Review by Schnurber — CHANGES_REQUESTED** [review]
Depending on the new version of tiled (open pull request), we can get rid of most `!`.

**Comment by sainoba** [comment]
Hi there, thank you for all the hard work.
Is there a timeline of when this PR will hit the pub.dev page??

**Comment by spydon** [comment]
Hi @sainoba, we just merged a big PR for tiled and that package has to be migrated to nullsafety first and then we can migrate this package, so it is probably some time left unfortunately.

**Comment by Schnurber** [comment]
@Akida31 you have to update the versions of tiled to `tiled: 0.7.0` and you have to create a pull request from your **dev** Branch not master!

**Comment by loothood** [comment]
Hi, thank you for the repository and for your hard work.
When can we use the null safety version of the package?

**Comment by Akida31** [comment]
Hey, I don't use flame currently so I won't develop this further. At the moment there is development on nullsafety in this repo: https://github.com/Schnurber/flame_tiled/tree/dev
This PR is basically only waiting for the other work and then it will be closed. It is still open to show that there is some work on this topic.

**Comment by Akida31** [comment]
Will be done with #10

**Pull request closed without merging**